python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/shape_lib_gen.py

```diff
@@ -1276,19 +1284,16 @@ def main(args):
 using namespace mlir;
 
 StringRef mlir::torch::Torch::getShapeLibrary() {{
-// TODO: Find a way to embed this string nicely.
-// It is currently too long, and will probably break MSVC builds if anyone
-// attempts that.
-// We want to preserve the legibility of the shape library as a checked in file,
-// since that is sometimes useful for debugging / diffing.
-// Probably the ideal outcome is to have the shape library be a .mlir file
-// that is checked in, and then we embed it as part of the build process.
+#ifndef _MSC_VER
 #pragma clang diagnostic push
```

Comment:
I don't think we need the pragmas now that we are doing the string line-wise

Reply:
That's fun. Clang must be considering the preprocessor-concatenated size rather than the individual string literal size, and MSVC only cares about the individual string literals? :rofl: